src/base/bittorrent/torrenthandle.cpp

```diff
@@ -961,12 +962,7 @@ bool TorrentHandle::hasMissingFiles() const
 
 bool TorrentHandle::hasError() const
 {
-#if (LIBTORRENT_VERSION_NUM < 10200)
-    return (m_nativeStatus.paused && m_nativeStatus.errc);
-#else
-    return ((m_nativeStatus.flags & lt::torrent_flags::paused)
-            && m_nativeStatus.errc);
-#endif
+    return static_cast<bool>(m_nativeStatus.errc);
```

Comment:
If `errc` can represent multiple error codes but only one non-error I would write it explicitly:
```c++
return m_nativeStatus.errc != 0;
```

Reply:
There is a `explicit operator bool() const` function in `error_code` class, that is why I wrote that way. So are you still sure about doing the comparison ourselves?
https://www.boost.org/doc/libs/1_71_0/boost/system/error_code.hpp

Reply:
It's good for using in `bool` contexts, IMO, e.g.:
```c++
if (m_nativeStatus.errc)
```
But in this case I would use comparison instead. I will not insist, after all...